[DO NOT MERGE] Replace Boost.Process with cpp-subprocess

hebasto opened the pull request:
Differences from https://github.com/bitcoin/bitcoin/pull/28981:

1. Dropped Windows-related changes, which allows to keep
2. Zero diff for the `subprocess.hpp`
3. Based on https://github.com/bitcoin/bitcoin/pull/29489

More details to be added soon :)

theStack committed:
external_signer: replace boost::process with cpp-subprocess

This primarily affects the `RunCommandParseJSON` utility function.

theStack committed:
build: remove boost::process dependency for building external signer support

hebasto committed:
test: Remove Windows-specific code from `system_tests/run_command`

This code has been dead since https://github.com/bitcoin/bitcoin/pull/28967.

Required as a precondition for replacing Boost.Process with
cpp-subprocess to make diff for this code meaningful and reviewable.

The plan is to reintroduce Windows-specific code in this test
simultaneously with enabling Windows support in cpp-subprocess.

fanquake commented:
Is there a reason this is a new PR, rather than pushing to the current one? Can the other one be closed?

hebasto commented:
> Is there a reason this is a new PR, rather than pushing to the current one?

To demonstrate the CI outcome for this approach to reviewers.

> Can the other one be closed?

I'll close it once this one is undrafted.

fanquake commented:
I don't really understand, because pushing to the other PR would also demonstrate the CI, and by switching to the PR, we'll now loose all the past discussion?

hebasto committed:
Add `cpp-subprocess` header-only library

Upstream repo: https://github.com/arun11299/cpp-subprocess
Commit: 4025693decacaceb9420efedbf4967a04cb028e7

The "Convenience Functions" section is unused in our codebase, so it has
been removed.
